src/main/java/seedu/address/logic/commands/SellCommand.java

```diff
@@ -0,0 +1,116 @@
+ */
+public class SellCommand extends Command {
+    public static final String COMMAND_WORD = "sell";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Sells the stated "
+            + "quantity of the good on the provided transaction date. "
+            + "Users are not allowed to sell more than they have in stock. "
+            + "Parameters: "
+            + PREFIX_GOOD_NAME + "GOOD_NAME "
+            + PREFIX_QUANTITY + "QUANTITY "
+            + PREFIX_TRANSACTION_DATE + "TRANSACTION_DATE "
+            + "Example: " + COMMAND_WORD + " "
+            + PREFIX_GOOD_NAME + "Apples "
+            + PREFIX_QUANTITY + "50 "
+            + PREFIX_TRANSACTION_DATE + "2020-01-01";
+
+    public static final String MESSAGE_SUCCESS = "Sold %1$d %2$ss";
+
+    private Good soldGood;
+
+    public SellCommand(Good soldGood) {
+        this.soldGood = soldGood;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        /*
+        if model doesn't have good, you cannot sell it
+
+        if model has good but not enough quantity, you cannot sell it
+
+        only if have good and sufficient qty then can sell
+         */
+        if (!model.hasGood(soldGood)) {
+            throw new CommandException("Trying to sell non-existent good");
+        }
+
+        if (!haveSufficientQuantity(model, soldGood)) {
+            throw new CommandException("Unable to sell a higher quantity "
+                    + "than amount in inventory");
+        }
+
+        decreaseQuantity(model, soldGood);
+        return new CommandResult(String.format(MESSAGE_SUCCESS,
+                soldGood.getGoodQuantity().goodQuantity, soldGood.getGoodName().fullGoodName));
+    }
+
+    /**
+     * Decreases the quantity of an existing good in inventory with the same name as {@code newGood}
+     * by the quantity in {@code newGood}
+     *
+     * @param model underlying model to make modifications in
+     * @param soldGood contains the good name and quantity to increase by
+     */
+    private void decreaseQuantity(Model model, Good soldGood) {
+        int oldGoodIndex = model.indexOfGood(soldGood);
+        Good oldGood = model.getFilteredGoodList().get(oldGoodIndex);
+
+        int updatedQuantity = oldGood.getGoodQuantity().goodQuantity
+                - soldGood.getGoodQuantity().goodQuantity;
+
+        Good updatedGood = new Good(new GoodName(soldGood.getGoodName().toString()),
+                new GoodQuantity(updatedQuantity));
+
+        model.setGood(oldGood, updatedGood);
+    }
+
+    /**
+     * Checks whether the inventory has at least the quantity of good trying to be sold.
+     * @param model underlying model whose inventory to check and mutate.
+     * @param soldGood name of good to sell and it's quantity.
+     * @return true if there is at least the quantity of good to be sold in the inventory.
+     */
+    private boolean haveSufficientQuantity(Model model, Good soldGood) {
```

Comment:
Should it be hasSufficientQuantity instead?

Reply:
Changed.